### PR TITLE
Make sure that blueprints with no slash is maintained when applied

### LINKED ISSUE #2080: bug: route without a trailing slash gets routed to route with a trailing slash, when the route is the top level of blueprint (maybe also app)
Version: 21.3.1

Description:

setting two urls in a blueprint with strict slashes for "" and "/" will raise an exception "route already registered" in sanic 21.3.1

In sanic version before 21.3 this would be acceptable

Code to reproduce:

```python
from sanic import Sanic, Blueprint
from sanic.response import text

app = Sanic("test")

bp = Blueprint("test", url_prefix="test")
@bp.get("", strict_slashes=True)
def _(req):
    return text("a")

@bp.get("/", strict_slashes=True)  # causes exception on sanic 21.3
def _(req):
    return text("b")

app.blueprint(bp)
app.run()
```

potential fix: https://github.com/sanic-org/sanic/pull/2079

### PR DESCRIPTION
**Pull request by ahopkins**
Resolves #2080
Replaces #2079

### CODE REVIEW

File: tests/test_routes.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 )
 from sanic_testing.testing import SanicTestClient
 
-from sanic import Blueprint, Sanic
+from sanic import Blueprint, Sanic, response
```
Comment (yunstanford) [inline tests/test_routes.py:15]: nit: redundant import ?

**Review by yunstanford — DISMISSED** [review]

File: tests/test_routes.py (hunk 1 above)
Comment (ahopkins) [reply tests/test_routes.py:15]: Thanks. Looks like VS Code was being a little over zealous on me.

**Comment by ashleysommer** [comment]
Not sure if this covers the condition when BP's (`url_prefix == ""` or `url_prefix is None`) and `url == ""`.

**Comment by ahopkins** [comment]
```python
bp = Blueprint("test")

@bp.get("")
def bar(req):
    return text("bar")
```

```
$ curl localhost:8888/ localhost:8888
bar
bar
```

**Pull request merged**